doc/api/stream.md

```diff
@@ -154,89 +325,134 @@ Examples of readable streams include:
+##### Event: 'close'
+
+The `'close'` event is emitted when the stream and any of its underlying
+resources (a file descriptor, for example) have been closed. The event indicates
+that no more events will be emitted, and no further computation will occur.
+
+Not all [Readable][] streams will emit the `'close'` event.
+
+##### Event: 'data'
 
 * `chunk` {Buffer|String} The chunk of data.
 
-Attaching a `'data'` event listener to a stream that has not been
-explicitly paused will switch the stream into flowing mode. Data will
-then be passed as soon as it is available.
+The `'data'` event is emitted when the stream is in flowing mode and there is
+data available to be consumed. Attaching a `'data'` event listener to a stream 
+that has not been explicitly paused will switch the stream into flowing mode. 
+Data will then be passed as soon as it is available.
 
-If you just want to get all the data out of the stream as fast as
-possible, this is the best way to do so.
+The listener callback will be passed the chunk of data as a string if a default
+encoding has been specified for the stream using the 
+`readable.setEncoding()` method; otherwise the data will be passed as a
+`Buffer`.
+
+The `'data'` event is the most efficient means of consuming all of a Readable
+streams data:
 
 ```js
-var readable = getReadableStreamSomehow();
+const readable = getReadableStreamSomehow();
 readable.on('data', (chunk) => {
-  console.log('got %d bytes of data', chunk.length);
+  console.log(`Received ${chunk.length} bytes of data.`);
 });
 ```
 
-#### Event: 'end'
+##### Event: 'end'
 
-This event fires when there will be no more data to read.
+The `'end'` event is emitted when there is no more data to be consumed from
+the stream.
 
-Note that the `'end'` event **will not fire** unless the data is
-completely consumed. This can be done by switching into flowing mode,
-or by calling [`stream.read()`][stream-read] repeatedly until you get to the
-end.
+*Note*: The `'end'` event **will not be emitted** unless the data is
+completely consumed. This can be accomplished by switching the stream into
+flowing mode, or by calling [`stream.read()`][stream-read] repeatedly until
+all data has been consumed.
 
 ```js
-var readable = getReadableStreamSomehow();
+const readable = getReadableStreamSomehow();
 readable.on('data', (chunk) => {
-  console.log('got %d bytes of data', chunk.length);
+  console.log(`Received ${chunk.length} bytes of data.`);
 });
 readable.on('end', () => {
-  console.log('there will be no more data.');
+  console.log('There will be no more data.');
 });
 ```
 
-#### Event: 'error'
+##### Event: 'error'
 
 * {Error}
 
-Emitted if there was an error receiving data.
-
-#### Event: 'readable'
+The `'error'` event will be emitted if an error occurred while receiving data.
+The listener callback will be passed a single `Error` object.
 
-When a chunk of data can be read from the stream, it will emit a
-`'readable'` event.
+##### Event: 'readable'
```

Comment:
Notably: this event isn't used by the `.pipe()` machinery. It exists to support the streams2-introduced pattern:

```js
function read (src, cb) {
  src.once('readable', onreadable);
  src.once('end', () => {
    return cb();
  })

  function onreadable () {
    while (1) {
      var data = src.read()
      if (data === null) {
        return;
      }
      // do something with data
    }
  }
}

```

Generally, users should prefer using `.pipe()` or `.on('data')`.